api/server/graph.cc

```diff
@@ -732,6 +738,74 @@ void Graph::NicConfigAntiSpoof(const app::Nic &nic, bool enable) {
     }
 }
 
+void Graph::EnablePacketTrace(const app::Nic &nic) {
+    Graph::GraphNic *g_nic = FindNic(nic);
+    std::string name;
+    if (nic.packet_trace) {
+        app::log.Debug("packet_trace has to be false on the nic to update!");
+        return;
+    }
+
+    if (g_nic->sniffer == NULL) {
+        name = "sniffer-" + g_nic->id;
+        g_nic->pcap_file = fopen(("/tmp/butterfly-" + std::to_string(getpid()) +
+                                 "-" + g_nic->id + ".pcap").c_str(), "w");
+        g_nic->sniffer = BrickShrPtr(pg_print_new(name.c_str(),
+                                 g_nic->pcap_file, PG_PRINT_FLAG_PCAP |
+                                 PG_PRINT_FLAG_CLOSE_FILE,
+                                 NULL, &app::pg_error),
+                       pg_brick_destroy);
+        if (!g_nic->sniffer) {
+            PG_ERROR_(app::pg_error);
+            return;
+        }
+    }
+    if (nic.bypass_filtering) {
+        unlink(g_nic->vhost);
+        g_nic->head = g_nic->sniffer;
+        link(g_nic->sniffer, g_nic->vhost);
+        link(vtep_, g_nic->head);
+        add_vni(vtep_, g_nic->head, nic.vni);
+    } else {
+        unlink_edge(g_nic->antispoof, g_nic->vhost);
+        g_nic->head = g_nic->sniffer;
+        link(g_nic->antispoof, g_nic->sniffer);
+        link(g_nic->sniffer, g_nic->vhost);
+    }
+}
+
+void Graph::DisablePacketTrace(const app::Nic &nic) {
+    Graph::GraphNic *g_nic = FindNic(nic);
+    if (!nic.packet_trace) {
+        app::log.Debug("packet_trace has to be true on the nic to update!");
```

Comment:
I don't find the message clear, would say as a WARNING:
> packet trace option on nic XXX is already disabled
Same for enabling
Also, we can avoid using "!"